admin/notifications/views.py

```diff
@@ -46,7 +50,11 @@ def process_duplicate_notifications(request):
 def handle_duplicate_notifications(request):
     detailed_duplicates, message, is_post = process_duplicate_notifications(request)
 
-    context = {'duplicates': detailed_duplicates}
+    paginator = Paginator(detailed_duplicates, 20)
```

Comment:
This paginator isn't saving any query overhead -- by the time it's initialized, `detect_duplicate_notifications` has already ran several hundred queries to produce `detailed_duplicates`.

In order to minimize query overhead, `detect_` would need to receive the page offset and serialize only the relevant notifications.